### PR TITLE
Fix #318 - ph5tostationxml does not handle stations with multiple deployments

### LINKED ISSUE #318: [BUG] ph5tostationxml does not handle stations with multiple deployments
**Describe the bug**
If a station had multiple deployments, then `ph5tostationxml` only returns metadata about the first deployment, instead of all deployments.

For example,

```
| => ph5tostationxml -n master.ph5 -p /PH5_Experiments/pn4-stage/19-004/ --station=9051 --format=text --level=channel
#Network|Station|Location|Channel|Latitude|Longitude|Elevation|Depth|Azimuth|Dip|SensorDescription|Scale|ScaleFreq|ScaleUnits|SampleRate|StartTime|EndTime
5B|9051||GP1|-78.13126749|-83.94024949|323.0|0.0|0.0|0.0|Geo Space GS-30CT/FairfieldNodal ZLAND 3C|2059290000.0|40.0|M/S|1000.0|2018-12-11T13:25:39|2018-12-11T17:21:53
5B|9051||GP2|-78.13126749|-83.94024949|323.0|0.0|90.0|0.0|Geo Space GS-30CT/FairfieldNodal ZLAND 3C|2059290000.0|40.0|M/S|1000.0|2018-12-11T13:25:39|2018-12-11T17:21:53
5B|9051||GPZ|-78.13126749|-83.94024949|323.0|0.0|0.0|90.0|Geo Space GS-30CT/FairfieldNodal ZLAND 3C|2059290000.0|40.0|M/S|1000.0|2018-12-11T13:25:39|2018-12-11T17:21:53
```

In the example above, station 9051 in experiment 19-004 had two deployments. The first deployment was from 2018-12-11T13:25:39  to 2018-12-11T17:21:53 while the second deployment was from 2018-12-28T18:52:19 to 2018-12-29T14:45:21. As you can see, only the deployment from 2018-12-11T13:25:39  to 2018-12-11T17:21:53 is included in the output.

**Additional context**
Experiment 19-004 is the first experiment where we have multiple deployments for a station. This fix is needed as soon as possible for a PI.

### PR DESCRIPTION
**Pull request by nick-iris**
### What does this PR do?
This PR provides a fix to issue #318. `ph5tostationxml` would only return first station & channel deployment for a station. This is a fix so that `ph5tostationxml` now returns all deployments.

### Relevant Issues?
#318 - [BUG] ph5tostationxml does not handle stations with multiple deployments

### Checklist
- [x] This PR is not directly related to an existing issue (which has no PR yet).
- [x] All tests pass.
- [x] Any new or changed features have are documented.
- [x] Changes have been added to `CHANGELOG.txt` .

### CODE REVIEW

**Comment by nick-iris** [comment]
@dsentinel Are you available to review this today? There is a PI that Alissa has been working with, and this fix addresses an issue required for us to archive their experiment.

**Comment by dsentinel** [comment]
I can give it a look, albeit with my limited knowledge of internals.

File: ph5/clients/ph5tostationxml.py [hunk 1]
```diff
@@ -658,69 +658,76 @@ def read_stations(self):
...
-                           sta_xml_obj.receiver_list != ['*']:
-                            obs_channels = self.read_channels(sta_xml_obj,
-                                                              station_list,
-                                                              deployment,
-                                                              station_code,
-                                                              array_code)
-                            # go to the next station if no channels were
-                            # returned
-                            if len(obs_channels) == 0:
+                        station_epochs = station_list[deployment]
```
Comment (dsentinel) [inline ph5/clients/ph5tostationxml.py]: This is bizarre. I know it's from before...
`station_list` must be a dictionary if it's being indexed on it's iterator :/

This is still redundant -> `for deployment, station_epochs in station_list.items()`

File: ph5/clients/ph5tostationxml.py [hunk 2]
```diff
@@ -658,69 +658,76 @@ def read_stations(self):
...
+                        for station_entry in station_epochs:
```
Comment (dsentinel) [inline ph5/clients/ph5tostationxml.py]: This function is too long and nested. Refactoring here may not be the place but you can reduce the nest (and all the line continuations) with
```
for deployment, station_epoch, station entry in ( (dk, dv, se) for dk, dv in station_list.items() for se in dv )
```

File: ph5/clients/ph5tostationxml.py (hunk 1 above)
Comment (dsentinel) [inline ph5/clients/ph5tostationxml.py]: Line 652 need to be executed twice?

File: ph5/clients/ph5tostationxml.py [hunk 3]
```diff
@@ -658,69 +658,76 @@ def read_stations(self):
...
+                            end_date = station_entry['pickup_time/epoch_l']
+                            end_date = UTCDateTime(end_date)
+                            if sta_xml_obj.start_time and \
+                                    sta_xml_obj.start_time > end_date:
+                                # chosen start time after pickup
+                                continue
+                            elif sta_xml_obj.end_time and \
+                                    sta_xml_obj.end_time < start_date:
+                                # chosen end time before pickup
+                                continue
```
Comment (dsentinel) [inline ph5/clients/ph5tostationxml.py]: For line coninuation, use `()` over `\` whenever you can.
```
                           if (sta_xml_obj.start_time and 
	                                    sta_xml_obj.start_time > end_date):
                            obs_station = self.create_obs_station(	    
...
```

File: ph5/clients/ph5tostationxml.py [hunk 4]
```diff
@@ -658,69 +658,76 @@ def read_stations(self):
...
+                            creation_date = start_date
+                            termination_date = end_date
+                            site_name = station_entry['location/description_s']
+
+                            obs_channels = []
+                            # run channel filters if necessary. we do this
+                            # first to avoid creating a station that has no
+                            # channels
+                            if self.manager.level.upper() == "RESPONSE" or \
```
Comment (dsentinel) [inline ph5/clients/ph5tostationxml.py]: also `()` over `/`

**Comment by nick-iris** [comment]
Thanks for the review @dsentinel. I cleaned up the code a bit based on your suggested changes. Please approve this request once you are satisfied.

**Comment by dsentinel** [comment]
> I think this was done for transparency, but I can remove these unnecessary variables if you find that to be cleaner.

Sorry I wasn't clear.
lat, long, etc deserve a variable. Leave those.
I was only referring to creation, and termination.

**Comment by nick-iris** [comment]
@dsentinel I added variables for latitude, longitude, and elevation back into the code. I'd like to merge once the test pass. Will you please approve the pull request if you think it looks okay?

**Pull request merged**